From 87b32ea6e4b4e34a8f899f86564be6c85f6335f7 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:37:48 -0400
Subject: [virt] kvm/vmx: count exits to userspace during invalid guest emulation

Message-id: <1406040016-3289-74-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86152
O-Subject: [RHEL7 PATCH v2 073/221] KVM: vmx: count exits to userspace during invalid guest emulation
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

These will happen due to MMIO.

Suggested-by: Gleb Natapov <gleb@redhat.com>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
Signed-off-by: Gleb Natapov <gleb@redhat.com>
(cherry picked from commit 94452b9e3401691c4d34a5a6f6a3a5b4e9c50a48)
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kvm/vmx.c b/arch/x86/kvm/vmx.c
index 8108127..963b866 100644
--- a/arch/x86/kvm/vmx.c
+++ b/arch/x86/kvm/vmx.c
@@ -5496,6 +5496,7 @@ static int handle_invalid_guest_state(struct kvm_vcpu *vcpu)
   err = emulate_instruction(vcpu, EMULTYPE_NO_REEXECUTE);
 
   if (err == EMULATE_USER_EXIT) {
+   ++vcpu->stat.mmio_exits;
    ret = 0;
    goto out;
   }
-- 
1.7.1